src/runtime-rs/src/shim/src/config.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2021 Alibaba Cloud
+// Copyright (c) 2021 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::path::{Path, PathBuf};
+use virtcontainers::config::TomlConfig;
+
+const SANDBOX_CONFIG_PATH_KEY: &str = "io.katacontainers.config_path";
+
+const DEFAULT_CONFIG_PATHS: &[&str] = &[
+    "/etc/kata-containers/configuration.toml",
+    "/usr/share/defaults/kata-containers/configuration.toml",
+];
+
+#[derive(thiserror::Error, Debug)]
+pub enum ConfigError {
+    #[error("failed to find configuration")]
+    Get,
+    #[error("failed to load configuration")]
+    Load(#[source] std::io::Error),
+    #[error("failed to parse configuration")]
+    Parse(#[source] toml::de::Error),
+}
+
+fn get_config_path_from_oci_spec(bundle_path: &Path) -> Option<String> {
+    let spec_file = bundle_path.join("config.json");
+    let spec = &oci_spec::runtime::Spec::load(spec_file).ok()?;
+
+    spec.annotations()
+        .as_ref()?
+        .get(SANDBOX_CONFIG_PATH_KEY)
+        .map(|v| v.into())
+}
+
+fn get_config_path_from_env() -> Option<String> {
+    std::env::var("KATA_CONF_FILE").ok()
```

Comment:
`KATA_CONF_FILE` should be a global `const`. It might be better to pass the variable name as a parameter to this function. That makes the test code cleaner since it doesn't need to set `KATA_CONF_FILE` (which might upset the environment the tests are running in).

Reply:
Change `KATA_CONF_FILE` to a global const in the first revision. But still directly use `KATA_CONF_FILE` in `get_config_path_from_env`. I think the env name also is part of the function logic